src/v/utils/rjson_allocator.h

```diff
@@ -0,0 +1,26 @@
+#include <rapidjson/allocators.h>
+#include <rapidjson/document.h>
+#include <rapidjson/encodings.h>
+
+namespace rjson_allocator {
+class MyAllocator : public rapidjson::CrtAllocator {
+public:
+    void* Malloc(size_t size) {
+        if (!size) {
+            return NULL;
+        }
+
+        void* res = rapidjson::CrtAllocator::Malloc(size);
+        if (NULL == res) {
+            throw std::bad_alloc{};
+        }
+
+        return res;
+    }
+};
```

Comment:
I think `CrtAllocator::Realloc` returns null on failure, should that be checked, too?

Reply:
added a wrapper for `CrtAllocator::Realloc`